### DISCUSSION TITLE
How to set golang version to 1.16

### DISCUSSION THREAD

**Discussion by Y4er** [vercel/vercel#6405] [Help]
I need to perform some other operations to build the article before the hugo command, like this: `go run main.go && hugo`

But the third-party library imported by my script must have the version of golang 1.16 or higher. How do I set the version of golang?

**Comment by barelyhuman** [comment]
From the docs. 
> The Go Runtime will automatically detect the go.mod file at the root of your Project to determine the version of Go to use.
>
> If go.mod is missing or the version is not defined, the default version 1.16 will be used.

Though I'm assuming you are using the hugo template by default and you might wanna run an extra go setup step since hugo doesn't need go and has the binary pre installed . 

you can add this command to the build step from the vercel > project > setting > general > build 

```sh
$ amazon-linux-extras install golang1.11 && go run main.go && hugo 
#or version specific 
$ amazon-linux-extras install golang1.16 && go run main.go && hugo 
```

Though the second command depends on the amazon linux extras to find the version 1.16 so it may or maynot work, the first command should go through.

You can find more info about this here : https://vercel.com/docs/build-step

**Reply by Y4er** [reply]
I need golang 1.16, but amazon-linux-extras does not have golang 1.16 version. So your command does not work.

**Reply by barelyhuman** [reply]
Okay, just to confirm that we have everything in the project as the runtime needs. 

1. You have `go.mod` file in the project? If not, `go mod init example_name`   
2. You’ve tested the main.go file with the module system on.

**Reply by Y4er** [reply]
1. i have go.mod in my project

![image](https://user-images.githubusercontent.com/40487319/124208894-2538a000-db1b-11eb-9401-51c1eaab9d31.png)

2. in my local env, it's working.

Golang installed through yum and amazon-linux-extras are all low versions, i want install golang1.16, is there any other easy way?

**Reply by barelyhuman** [reply]
The reason for those questions is that vercel states to install 1.16 by default if the go.mod states no version. And if that’s not happening, I’d like to know what error you are getting when building.
